nixos/modules/system/boot/systemd-unit-options.nix

```diff
@@ -209,10 +237,16 @@ in rec {
     };
 
     script = mkOption {
-      type = types.lines;
-      default = "";
-      description = "Shell commands executed as the service's main process.";
-    };
+        type = with types; either lines (submodule subscript);
+        default = "";
+        description = ''
+          Code executed as the service's main process.
```

Comment:
Would it be unreasonable to simply include a shebang line at the top of the script segment, rather than adding a bunch of plumbing to explicitly support each language?

e.g. would `#!${haskellPackages.ghc}/bin/runghc` be good enough?